agent/hcp/client/metrics_client.go

```diff
@@ -150,8 +154,18 @@ func (o *otlpClient) ExportMetrics(ctx context.Context, protoMetrics *metricpb.R
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return fmt.Errorf("failed to export metrics: code %d: %s", resp.StatusCode, string(body))
+		truncatedBody := truncate(respData.String(), defaultErrRespBodyLength)
+		return fmt.Errorf("failed to export metrics: code %d: %s", resp.StatusCode, truncatedBody)
 	}
 
 	return nil
 }
+
+func truncate(text string, width uint) string {
```

Comment:
width can just be an int?

Reply:
Definitely but I didn't want to add a check to see if it was less than 0 hence the uint. I can do so if you'd prefer but I though this was more human readable.

Reply:
my take is both could be argued as more readable:
- uint because it removes the extra code to check, reduces cyclomatic complexity, etc.
- int because it is more familiar and these checks are quite common.

I would probably lean to the int due to familiarity, but its only a nit that I would block on or a hill that I would die on.